Extensions/TextObject/textruntimeobject.js

```diff
@@ -3,46 +3,111 @@
  *  2013-2016 Florian Rival (Florian.Rival@gmail.com)
  */
 
+/**
+ * @typedef {Object} textObjectDataType Base parameters for gdjs.TextRuntimeObject
+ * @property {number} characterSize The size of the characters
+ * @property {string} font The font name
+ * @property {boolean} bold Is Bold?
+ * @property {boolean} italic Is Italic?
+ * @property {boolean} underlined Is Underlined?
+ * @property {Object} color The text color in an RGB representation
+ * @property {number} color.r The Red level from 0 to 255
+ * @property {number} color.g The Green level from 0 to 255
+ * @property {number} color.b The Blue level from 0 to 255
+ * @property {string} string The text of the object
+ * 
+ * @typedef {objectData & textObjectDataType} textObjectData
+ */
+
 /**
  * Displays a text on the screen.
  *
  * @memberof gdjs
  * @class TextRuntimeObject
  * @extends RuntimeObject
+ * @param {gdjs.runtimeScene} runtimeScene The parent gdjs.RuntimeScene Instance
+ * @param {textObjectData} textObjectData The Optional object Data
```

Comment:
Careful about the casing in your description: "The object data used to initialize the object"

It's also not optional :)